core/settlement/engine.go

```diff
@@ -476,6 +476,13 @@ func (e *Engine) AddPosition(party string, evt events.MarketPosition) {
 	_ = e.getCurrentPosition(party, evt)
 }
 
+func (e *Engine) HasPosition(party string) bool {
+	_, okPos := e.pos[party]
+	_, okTrades := e.trades[party]
```

Comment:
I don't understand the okTrades - this would return true in all cases you ever had a position and were not distressed, even if you don't hold it anymore. what is the purpose of it?

Reply:
well it's not always that there's a position AND trades during the MTM window. you could have a position (which in that case was registered with a potential position at the time) but 0 trades. but at the end of the window, the party might have cancelled their orders and no trades happened. At that point the position engine might have deleted the position as it's of size 0, and no orders are left in the book.